tests/nmap_agent_test.py

```diff
@@ -1214,3 +1316,22 @@ def testAgentNmapOptions_whenServiceHasNoProduct_reportsFingerprintzzz(
     nmap_test_agent.process(domain_msg)
 
     assert any("fingerprint" in msg.selector for msg in agent_mock) is False
+
+
+def testAgentNmapOptions_whenNmaprunHostIsList_noCrash(
+    nmap_test_agent: nmap_agent.NmapAgent,
+    agent_mock: List[message.Message],
+    agent_persist_mock: Dict[Union[str, bytes], Union[str, bytes]],
+    domain_msg: message.Message,
+    mocker: plugin.MockerFixture,
+) -> None:
+    """Test host in scan results is a list instead of a dict, the agent should not
+    crash, instead it completes its emitting process for each of the hosts in the host list."""
+    mocker.patch(
+        "agent.nmap_wrapper.NmapWrapper.scan_domain",
+        return_value=(SCAN_RESULT_HOST_AS_LIST, HUMAN_OUTPUT),
+    )
+
+    nmap_test_agent.process(domain_msg)
+
+    assert len(agent_mock) == 16
```

Comment:
Can you please add explicit checks on the messages.

Reply:
+1

Reply:
Yes, please assert on the content.